rest/src/main/java/io/syndesis/rest/v1/handler/connection/ConnectionActionHandler.java

```diff
@@ -58,4 +84,38 @@ public ActionDefinition definition(@PathParam("id") @ApiParam(required = true,
         return action.map(Action::definition).orElseThrow(() -> new EntityNotFoundException("Action with id: " + id));
     }
 
+    @GET
+    @Path(value = "/{id}/properties")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation("Retrieves action property suggestions")
+    @ApiResponses(@ApiResponse(code = 200, response = ActionPropertySuggestions.class,
+        message = "A map of zero or more action property suggestions keyed by the property name"))
+    public ActionPropertySuggestions properties(
+        @PathParam("id") @ApiParam(required = true,
+            example = "io.syndesis:salesforce-create-or-update:latest") final String id,
+        @Context final UriInfo uriInfo) {
+        final String connectorId = connector.getId().get();
+        final Action action = connector.actionById(id).orElseThrow(
+            () -> new EntityNotFoundException("No Action with id: " + id + ", found for connector: " + connectorId));
+
+        // put all specified parameters in query parameters
+        final MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+        final Map<String, Object> parameters = queryParameters.entrySet().stream()
+            .collect(Collectors.toMap(Entry::getKey, e -> e.getValue().get(0)));
+
+        // put all action parameters with `null` values
+        action.definition().propertyDefinitionSteps()
+            .forEach(step -> step.getProperties().forEach((k, v) -> parameters.putIfAbsent(k, null)));
+
+        // lastly put all connection properties
+        parameters.putAll(connection.getConfiguredProperties());
+
+        final Client client = ClientBuilder.newClient();
+        final WebTarget target = client
+            .target(String.format("http://%s/api/v1/action/properties/%s", config.getService(), connectorId));
+
+        return target.request(MediaType.APPLICATION_JSON)
```

Comment:
I know this is the simplest way to get data from the syndesis-verifier (we should really check for a new name ;-) to the UI but it also tightly couples syndesis-ui to syndesis-verifier via this data type (ActionPropertySuggestions). The same problem we already have for the validation calls and which already leads to some strange constructs in the client (there is also an issue open for this to fix).

IMO it would be much better do decouple those (e.g. that the syndesis-verifier has not direct or indirect dependency on syndesis-rest) and let the verifier stay in charge of the model object it serves (and uses as request parameters).

Haven't yet looked in the verifier code, but the dependency chain (including 'copied' objects) should be 
`syndesis-ui --> syndesis-rest --> syndesis-verifier` and it should be unidirectional (i.e. no circles)

Reply:
Just have looked at the corresponding syndesis-verifier PR (https://github.com/syndesisio/syndesis-verifier/pull/18) and it seems that there is no directly coupling between object (but with the expected data format so that it can be serialized on both sides). 

I now just would wait on feedback from the UI what is the best way to consume the data and if there are changes required we can put the transformation into this here.